docs/message-queues.asciidoc

```diff
@@ -0,0 +1,98 @@
+[[message-quques]]
+== Message Queues
+
+The Node.js Agent will automatically create spans for any activity to/from your Amazon SQS message queues.  In order to have those spans recorded, your message queue activity must  occur during a transaction.
+
+If you're performing queue operations during an HTTP request from a <<compatibility-frameworks,supported framework>>     the agent will start a transaction automatically.  However, if you're performing queue operations in a stand alone program (such as a message processor) you'll need to use the Node.js Agent's <<apm-start-transaction,`startTransaction()`>> method to create transactions for your messages.
+
+You can see an example of this in the following code sample.
+
+[source,js]
+----
+const apm = require('elastic-apm-node').start({/*...*/})
+const AWS = require('aws-sdk');
+// Set the region
+AWS.config.update({region: 'us-west'});
+
+// Create an SQS service object
+const sqs = new AWS.SQS({apiVersion: '2012-11-05'});
+
+/* ... */
+
+const transaction = apm.startTransaction("Process Messages", 'cli')
+sqs.receiveMessage(params, function(err, data) {
+  if(err) {
+    console.log("Receive Error", err);
+  } else {
+    console.log(`Length: ${data.Messages.length}`)
+    /* process messages */
+  }
+  // end the transaction
+  transaction.end()
+})
+----
+
+In the above code you can see we've started a transaction before calling the `sqs.receiveMessage` method, and ended it _after_ the queue's processing callback finishes executing.  The will ensure a transaction is active while processing your queue messages.
+
+[float]
+[[message-queues-distributed-tracing]]
+=== Distributed Tracing and Messaging Queues
+
+Similarly, if you want to connect your queue scheduling and queue processing via distributed tracing, you'll need to use the Node.js Agent's API to _store_ a `traceparent` header with your queue message and then provide that `traceparent` header when starting the new transaction.
+
+Here's an example that uses the Node.js Agent API to store the `traceparent` as a message attribute and then uses that attribute to link your new transaction with the original.
+
+First, when sending the message, we add the trace as one of the `MessageAttributes`
+[source,js]
+----
+// stores the traceparent when sending the queue message
+const traceParent = apm.currentTransaction ? apm.currentTransaction.traceparent : ''
+
+// Use the Amazon SQS `MessageAttributes` to pass 
+// on the  traceparent header
+const params = {
+  /* ... other params ... */
+  MessageAttributes: {
+    /* ... other attributes ... */
+    "MyTraceparent":{
+        DataType: "String",
+        StringValue: traceParent
+    }
+  }
+
+}
+sqs.sendMessage(params, function(err, data) {
+  /* ... */
+});
+----
+
+Then, when receiving the messages, we start and end our transaction while looping over the messages.  
```

Comment:
I'm confused here. The previous text only refers to a singular message, but now we're talking about multiple messages:

> First, when sending the message, we add the trace as one of the `MessageAttributes`

> Then, when receiving the messages, we start and end our transaction while looping over the messages.

Reply:
appologies -- looks like a copy/pasta error with the code sample.  I will correct

Reply:
@bmorelli25 I think I see the confusion here. It was not copy/pasta as I originally thought.  

The SQS API _schedules_ a single message at a time.  However, when you're consuming messages, the API always returns an array of messages.  My original text assumed the reader had some experience with SQS queues already and would have internalized this. 

I've made some small adjustments to the section to address this, and the other place I introduced confusion.  If this is still a head scratcher lets sync up via slack or zoom to talk it out.